ext/liquid_c/block.c

```diff
@@ -78,6 +101,18 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             {
                 const char *start = token.str + 2, *end = token.str + token.length - 2;
 
+                if (token.str[2] == '-') {
+                    if (RARRAY_LEN(nodelist) > 0) {
+                        VALUE last_node = RARRAY_AREF(nodelist, RARRAY_LEN(nodelist)-1);
+                        rb_funcall(last_node, intern_rstrip, 0);
+                    }
+                    start++;
+                }
+
+                if (token.str[token.length - 3] == '-') {
```

Comment:
I also think we should avoid this code duplication using an inline function so we can solve this problem once.  This function could also be defined with the tokenizer code so whitespace trimming can be handled in one place.  There could be one function to get the tag token contents (i.e. without the tag start, end or trim characters) and another to get the raw token contents (i.e. after trimming).